ember-core/shared/src/main/scala/org/http4s/ember/core/h2/H2Error.scala

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2019 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s.ember.core.h2
+
+import cats.syntax.all._
+private[h2] sealed abstract class H2Error(val value: Int) {
+  def toGoAway(highest: Int): H2Frame.GoAway =
+    H2Frame.GoAway(0, highest, value, None)
+  def toRst(stream: Int): H2Frame.RstStream =
+    H2Frame.RstStream(stream, value)
+
+}
+private[h2] object H2Error {
+
+  def fromInt(int: Int): Option[H2Error] = int match {
```

Comment:
Does this happen enough to merit caching the somes?

Reply:
Not generally I don't think.